src/java/org/apache/cassandra/guardrails/GuardrailsConfig.java

```diff
@@ -89,8 +89,8 @@
 
     public volatile Long tables_warn_threshold;
     public volatile Long tables_failure_threshold;
-    public volatile ImmutableSet<String> table_properties_disallowed;
-    public volatile ImmutableSet<String> table_properties_ignored;
+    public volatile Set<String> table_properties_disallowed;
```

Comment:
But why did you revert that? I was ok with ignoring that problem in Sonar, though using immutable set here for a volatile field was imho a good idea - otherwise it would have to be a concurrent set if we are considering thread safety

Reply:
Sorry, I thought I had made a comment about that.

I reverted because during a dtest I was seeing this error and C* didn't start. I think the `YamlConfigurationLoader` is trying to set the value using `Collections.setFromMap`. Changing back to `Set` got things working again.

```
INFO  [main] 2021-06-03 11:43:30,948 YamlConfigurationLoader.java:93 - Configuration location: file:/private/var/folders/bp/dmn9bb6d09v8fwpmts32xftm0000gn/T/dtest-nkyslfbx/test/node1/conf/cassandra.yaml
ERROR [main] 2021-06-03 11:43:31,657 CassandraDaemon.java:909 - Exception encountered during startup: Invalid yaml: file:/private/var/folders/bp/dmn9bb6d09v8fwpmts32xftm0000gn/T/dtest-nkyslfbx/test/node1/conf/cassandra.yaml
 Error: Cannot create property=guardrails for JavaBean=org.apache.cassandra.config.Config@3ee37e5a
 in 'reader', line 1, column 1:
    allocate_tokens_for_local_replic ...
    ^
Cannot create property=write_consistency_levels_disallowed for JavaBean=org.apache.cassandra.guardrails.GuardrailsConfig@2805d709
 in 'reader', line 54, column 3:
      tombstone_failure_threshold: 500
      ^
Can not set com.google.common.collect.ImmutableSet field org.apache.cassandra.guardrails.GuardrailsConfig.write_consistency_levels_disallowed to java.util.Collections$SetFromMap
 in 'reader', line 56, column 40:
     ... _consistency_levels_disallowed: {}
                                         ^

 in 'reader', line 54, column 3:
      tombstone_failure_threshold: 500
      ^
```

Reply:
ok, so maybe I'm extremely picky there, but this seems to be wrong - I mean, since we assume the configuration object is going to be modified in a concurrent way (which I guess is the reason why we have `volatile` here and there), having a simple set, which is not thread-safe and allows for modification is wrong

Reply:
Why did these all become `volatile`?  They are not in DSE.

Reply:
It was decide to leave the `volatile Set`'s since they are only used externally by cndb code.

Reply:
> Why did these all become `volatile`? They are not in DSE.

Discussed on zoom; for reference here, it was changed for cndb in [cndb-1842](https://github.com/riptano/cndb/pull/1878)